src/pages/index.js

```diff
@@ -22,128 +23,270 @@ import {
+      .catch(console.error);
+  } else {
+    return api
+      .removeLike(cardId)
+      .then(() => false)
+      .catch(console.error);
+  }
+}
+
+//// Button Event Listeners ////
+// avatar image click
+profileAvatarWrapper.addEventListener("click", openEditAvatarModal);
+
 // profile edit button click
 profileEditButton.addEventListener("click", openEditProfileModal);
 
 // add card button click
 addCardButton.addEventListener("click", openAddCardModal);
 
-//// Forms ////
-const profileForm = document.forms["edit-profile-form"];
-const addCardForm = document.forms["add-card-form"];
+//// Form Submit Handlers ////
+function handleSubmit(request, popupInstance, loadingText = "Saving...") {
+  // change the button text
+  popupInstance.renderLoading(true, loadingText);
+  request()
+    .then(() => {
+      popupInstance.close();
+    })
+    .catch(console.error)
+    .finally(() => {
+      popupInstance.renderLoading(false);
+    });
+}
 
-//// Form Input Fields ////
-const profileNameInput = profileForm.querySelector(".modal__input_type_name");
-const profileDescriptionInput = profileForm.querySelector(
-  ".modal__input_type_description"
-);
-const newCardTitleInput = addCardForm.querySelector(".modal__input_type_title");
-const newCardUrlInput = addCardForm.querySelector(".modal__input_type_url");
+// Avatar form submission handler
+function handleAvatarFormSubmit(data) {
+  function makeRequest() {
+    return api.updateUserAvatar(data).then((profileAvatar.src = data.link));
+  }
+  handleSubmit(makeRequest, editAvatarPopup);
+}
 
-//// Form Submit Handlers ////
 // profile form submission handler
 function handleProfileFormSubmit(data) {
-  // insert new values into the textContent property of the
-  // corresponding profile elements
-  user.setUserInfo(data);
-
-  // // close popup
-  // profilePopup.close();
-  // profilePopup.reset();
+  // change propert name of description to about
+  const updatedData = { name: data.name, about: data.description };
+
+  function makeRequest() {
+    return api.updateUserInfo(updatedData).then((userData) => {
+      user.setUserInfo({
+        name: userData.name,
+        about: userData.about,
+        avatar: userData.avatar,
+      });
+    });
+  }
+
+  handleSubmit(makeRequest, profilePopup);
 }
 
 // Add card form submission handler
 function handleAddCardFormSubmit(data) {
-  // create new card
-  const newCard = createCard(data, openImageModal);
-
-  // Add new card to begining of card gallery
-  cardsSection.addItem(newCard);
-
-  addCardPopup.close();
-  addCardPopup.reset();
-  formValidators[addCardForm.getAttribute("id")].toggleButtonState();
+  function makeRequest() {
+    // add new card to server
+    return api.addNewCard(data).then((cardData) => {
+      // create new card with the data returned from server
+      const newCard = createCard(cardData);
+
+      // Add new card to begining of card gallery
+      cardsSection.addItem(newCard);
+
+      addCardPopup.close();
```

Comment:
`  addCardPopup.close();` is not needed because you have everything inside `handleSubmit`